src/PrestaShopBundle/Model/Product/AdminModelAdapter.php

```diff
@@ -532,9 +532,15 @@ function ($p) {
      */
     private function mapStep2FormData(Product $product)
     {
+        // ecotax is stored with tax included but form uses the tax excluded value
+        $ecotax = $this->tools->round(
+            $product->ecotax * (1 + $this->taxRuleDataProvider->getProductEcotaxRate() / 100),
+            2
```

Comment:
Could you use a constant instead? or explain why you use 2 in a code comment.

Reply:
Initially I was copying the behavior of `AdminProductsController::_applyTaxToEcotax()` but I see that indeed I should rather pick the same precision that is being used in the function that computes "tax excl." from "tax. incl." price, which is `AdminProductsController::_removeTaxFromEcotax()`

So, as `AdminProductsController::_removeTaxFromEcotax()` uses 6 precision digits, I will use 6 too.

Reply:
Fixed in https://github.com/PrestaShop/PrestaShop/pull/18580/commits/f3b1d0e650165f0b4b61959deda23a999c346c21